.github/workflows/update-python-stubs.yml

```
@@ -0,0 +1,68 @@
+name: Auto-update PyO3 stubs
+on:
+  pull_request:
+    branches: [main] # PRs _into_ main
+    paths:
+      - "rust/kcl-python-bindings/**" # run only when Rust/Python code changes
+      - ".github/workflows/update-python-stubs.yml"
+permissions:
+  contents: write # allow committing back to the PR branch
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+jobs:
+  build-and-stub:
+    runs-on: ubuntu-latest
```

If the jobs still work, we'd prefer everything to use Namespace runners.

```suggestion
    runs-on: namespace-profile-ubuntu-2-cores
```